wgpu-core/src/device/resource.rs

```diff
@@ -2304,10 +2326,13 @@ impl<A: HalApi> Device<A> {
 
         // validate total resource counts
         for &id in desc.bind_group_layouts.iter() {
-            let bind_group_layout = bgl_guard
-                .get(id)
-                .map_err(|_| Error::InvalidBindGroupLayout(id))?;
-            count_validator.merge(&bind_group_layout.count_validator);
+            let bind_group_layout = match try_get_bind_group_layout(bgl_guard, id) {
+                Some(layout) => layout,
+                None => {
+                    return Err(Error::InvalidBindGroupLayout(id));
+                }
+            };
```

Comment:
nitpick: We could use `let-else` here:

```suggestion
            let Some(bind_group_layout) = try_get_bind_group_layout(bgl_guard, id) else {
                return Err(Error::InvalidBindGroupLayout(id));
            };
```

Pushed as a `fixup!` with 77b37c98.